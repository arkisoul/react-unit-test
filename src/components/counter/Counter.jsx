import { useState } from "react";

const Counter = (props) => {
    const [username, setUsername] = useState('hello')

    return (
      <div className="counter">
        <h1 className="count">{props.count}</h1>
        <p className="name">{username}</p>
        <button onClick={() => setUsername("greet")}>Set name</button>
      </div>
    );
}

export default Counter;