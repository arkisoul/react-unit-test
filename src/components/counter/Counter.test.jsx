import {shallow} from 'enzyme';

import Counter from './Counter'

describe('test suit for counter component', () => {
    const counter = shallow(<Counter />)
    it('should render properly', () => {
        expect(counter.getElements()).toMatchSnapshot()
    })
    
    it('should have a heading (h1) tag', () => {
        expect(counter.find('h1.count').length).toEqual(1)
    })
    
    it('should have text 4 inside h1 tag', () => {
        expect(counter.setProps({count: 4}).find('h1.count').text()).toEqual("4")
        expect(counter.setProps({count: 8}).find('h1.count').text()).toEqual("8")
    })

    it('should have a name state', () => {
        expect(counter.username).toEqual("hello");
        counter.find('button').simulate('click')
        expect(counter.username).toEqual("greet");
    })
})