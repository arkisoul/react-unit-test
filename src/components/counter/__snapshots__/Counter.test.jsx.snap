// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suit for counter component should render properly 1`] = `
Array [
  <div
    className="counter"
  >
    <h1
      className="count"
    />
    <p
      className="name"
    >
      hello
    </p>
    <button
      onClick={[Function]}
    >
      Set name
    </button>
  </div>,
]
`;
